Woof/Woof/ViewModel/LoginViewModel.swift

```diff
@@ -0,0 +1,12 @@
+import Foundation
+
+/// An interface providing for `the LoginView` to interact with the model layer.
+final class LoginViewModel: ObservableObject {
+    /// Requests the model layer to save the selected user role.
+    ///
+    /// - Parameter userRole: The role to be saved.
+    /// - Returns: A boolean value indicating whether the operation was successful.
+    func set(_ userRole: Role) -> Bool {
```

Comment:
How will we pass the `Role` to the view model? Do we need to call this function in the `LoginViewView`?

Reply:
in LoginView we could use this method:
``` swift
   var viewModel = LoginViewModel()
 
   Button(action: {viewModel.set(.owner)}, label: {})
```

Reply:
You pass the model object in the view. It violates MVVM architecture.

Reply:
The team decided to refactor the code to mvvm later